src/RequestHandlerSwooleRunner.php

```diff
@@ -193,24 +206,102 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
+        $this->logger = $logger ? : new StdoutLogger();
+        $this->pidManager = $pidManager;
     }
 
     /**
      * Run the application
      */
     public function run() : void
     {
-        $this->swooleHttpServer->on('start', [$this, 'onStart']);
-        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
-        $this->swooleHttpServer->start();
+        $opts = new Getopt([
+            'daemonize|d'  => 'daemonize the swoole server process',
+            'worker_num|n=i' => 'set worker nums',
+            'dispatch_mode=i' => 'set dispatch_mode',
+        ]);
+        $args = $opts->getArguments();
+        $action = $args[0] ?? null;
+        switch ($action) {
+            case 'stop':
+                $this->stopServer();
+                break;
+            case 'start':
+            default:
+                $this->startServer([
+                    'daemonize' => $opts->getOption('daemonize') ? (bool)$opts->getOption('daemonize') : false,
+                    'dispatch_mode' => $opts->getOption('dispatch_mode') ? (int)$opts->getOption('dispatch_mode') : 3,
+                    'worker_num' => $opts->getOption('worker_num') ? (int)$opts->getOption('worker_num') : 1,
+                ]);
+                break;
+        }
+    }
+
+    /**
+     * Start swoole server
+     *
+     * @param array $options Swoole server options
+     */
+    public function startServer(array $options = [])
+    {
+        $swooleServer = $this->server->createSwooleServer($options);
+        $swooleServer->on('start', [$this, 'onStart']);
+        $swooleServer->on('request', [$this, 'onRequest']);
+        $swooleServer->start();
+    }
+
+    /**
+     * Stop swoole server
+     */
+    public function stopServer()
+    {
+        if (! $this->isRunning()) {
+            $this->logger->info('Server is not running yet');
+            return;
+        }
+        [$masterPid, ] = $this->pidManager->read();
+        $this->logger->info('Server stopping ...');
+        $result = SwooleProcess::kill((int)$masterPid);
+        $startTime = time();
+        while (! $result) {
+            if (SwooleProcess::kill((int)$masterPid, 0)) {
+                if (time() - $startTime >= 60) {
+                    $result = false;
+                    break;
+                }
+                usleep(10000);
+                continue;
+            }
+        }
+        if (! $result) {
+            $this->logger->info('Server stop failure');
+        } else {
+            $this->pidManager->delete();
+            $this->logger->info('Server stopped');
+        }
+    }
+
+    /**
+     * Is swoole server running ?
+     */
+    public function isRunning() : bool
+    {
+        [$masterPid, $managerPid] = $this->pidManager->read();
+        if ($managerPid) {
+            // Swoole process mode
+            return $masterPid && $managerPid && SwooleProcess::kill((int)$managerPid, 0);
+        } else {
```

Comment:
There's no need for an `else` clause, as the `if` body returns; remove it, and promote the contents to the main method body.